dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/event/MetricsEventMulticaster.java

```diff
@@ -21,7 +21,11 @@
 
 public interface MetricsEventMulticaster {
 
-    void addListener(MetricsListener listener);
+    void addListener(MetricsListener<?> listener);
 
-    void publishEvent(MetricsEvent event);
+    void publishEvent(MetricsEvent<?> event);
+
+    void publishFinishEvent(MetricsEvent<?> event);
```

Comment:
Why is the publish event split into 3 methods? Can it be distinguished by MetricsEvent.Type

Reply:
Use eventMulticaster.publishEvent(FinishRegisterEvent) or eventMulticaster.publishFinishEvent(RegisterEvent) to indicate the end of a certain type of event, but the metric buried point scene has three fixed steps. I think publishFinishEvent is more descriptive of the life cycle